src/main/java/com/liveramp/cascading_ext/yarn/YarnApiHelper.java

```diff
@@ -76,14 +79,62 @@ private static String GETRequest(String urlString) throws IOException {
   }
 
   public static Optional<ApplicationInfo> getYarnAppInfo(Configuration conf, String appId) {
-    String yarnApiAddress = conf.get("yarn.resourcemanager.webapp.address", "");
-    if ( !yarnApiAddress.isEmpty()) {
+    String yarnApiAddress = getYarnApiAddress(conf);
+    if (!yarnApiAddress.isEmpty()) {
       return getYarnAppInfo(yarnApiAddress, appId);
     } else {
       return Optional.empty();
     }
   }
 
+  /***
+   * Gets a yarn api address. This also works if you are using more than one resource manager.
+   * If there's more than one resource manager this method will return the address of the first
+   * resource manager with which it was able to successfully connect to. It will return an empty
+   * string if there were no resource manager it could successfully connect to.
+   */
+  private static String getYarnApiAddress(Configuration conf) {
```

Comment:
If you used an `Optional` here (which represents the notion that you may not get an RM), you could turn [these lines](https://github.com/LiveRamp/cascading_ext/pull/90/files#diff-5796ba38ae3b31ff5eab085bca3465e8R83-R89) into 

```
getYarnApiAddress(conf).map(x -> getYarnAppInfo(x, appId))
```